src/solvers/mp/rcspp/SubProblem.h

```diff
@@ -0,0 +1,381 @@
+
+		// 3 -> [No short + barely above legal]
+		//		short = false,
+		//		max   = CD_max+1
+		//		sink  = one / last day
+		//
+		case 3: shortRotationsStrategy_=2; maxRotationLength_+=1; oneSinkNodePerLastDay_ = true; break;
+
+		// UNKNOWN STRATEGY
+		default:
+			std::cout << "# Unknown strategy for the subproblem (" << strategy << ")" << std::endl;
+			break;
+		}
+	}
+
+	// *** PARAMETERS ***
+
+	// 0 -> no short rotations
+	// 1 -> all short rotations
+	// 2 -> day-0 and last-day short rotations only
+	// 3 -> day-0 short rotations only
+	// 4 -> last-day short rotations only
+	int shortRotationsStrategy_ = 0;
+
+	// maximal length for a rotation
+	int maxRotationLength_ = 0;
+
+	// true  -> one sink node per day
+	// false -> one single sink node for the network
+	bool oneSinkNodePerLastDay_ = false;
+
+	// Getters for the class fields
+	//
+	int maxRotationLength(){ return maxRotationLength_; }
+	int shortRotationsStrategy(){return shortRotationsStrategy_;}
+	bool oneSinkNodePerLastDay(){return oneSinkNodePerLastDay_;}
+
+	// Setters
+	//
+	void maxRotationLength(int value){maxRotationLength_ = value;}
+	void shortRotationsStrategy(int value){shortRotationsStrategy_ = value;}
+	void oneSinkNodePerLastDay(bool value){oneSinkNodePerLastDay_ = value;}
+
+
+};
+
+
+//---------------------------------------------------------------------------
+//
+// C l a s s   S u b P r o b l e m
+//
+// Contains the shortest paths with resource constraints
+//
+//---------------------------------------------------------------------------
+
+class SubProblem {
+
+  public:
+
+    SubProblem();
+
+    virtual ~SubProblem();
+
+    // Constructor that correctly sets the resource (time + bounds), but NOT THE COST
+    //
+    SubProblem(Scenario* scenario, int nbDays, PConstContract contract, std::vector<State> *pInitState);
+
+    // Initialization function for all global variables (not those of the rcspp)
+    //
+    virtual void init(const std::vector<State> &pInitState);
+
+    // Solve : Returns TRUE if negative reduced costs path were found; FALSE otherwise.
+    //
+    virtual bool solve(LiveNurse*  nurse,
+                       DualCosts *costs,
+                       SubproblemParam param,
+                       std::set<std::pair<int, int> > forbiddenDayShifts = {},
+                       std::set<int> forbiddenStartingDays = {},
+                       bool optimality = false,
+                       double redCostBound = -1e-9);
+
+    // Returns all rotations saved during the process of solving the SPPRC
+    //
+    inline const std::vector<RCSolution>& getSolutions() const { return theSolutions_; }
+
+    virtual void build();
+
+    // Some getters
+    //
+    inline Scenario* scenario() const { return pScenario_; }
+
+    inline PConstContract contract() const { return pContract_; }
+
+    inline const LiveNurse*  liveNurse() const { return pLiveNurse_; }
```

Comment:
With the roster decomposition, it seems that the only difference between the subproblems of nurses having the same contract will be on preferences. It seems that it may really become useful to solve for one contract (possibly without considering the preferences) and then price the solutions for each nurse (I mean as a heuristic).
 This may even be more important in a stochastic approach if we do not generate random preferences...

Reply:
There is also the issue with historical state which is more important

Reply:
agreed, I wrote this comment without thinking long enough. I think I must be frustrated I never actually looked if new columns could be interesting for more than one nurse at a time.I think I might still add an an issue as a reminder to myself. I simply need to convince myself this kind of approach is useless.